app/routes/products_.$productId/route.tsx

```diff
@@ -41,8 +42,16 @@ export default function ProductDetailsPage() {
         if (!product?._id) {
             return;
         }
-        const quantity = parseInt(quantityInput.current?.value || '1', 10);
 
+        setSubmitClicked(true);
+        if (
+            product.productOptions &&
+            product.productOptions.some((c) => selectedOptions[c.name || ''] === undefined)
```

Comment:
|| -> ??

Reply:
so, I saw there are color options, dropdown options and some other types of options. You are only rendering color and dropdown options. But this `some` loop is checking every option.
Maybe we need to create an array or supported options from a product and to only work with that array ?

Reply:
1. fixed
2. this is not correct, there are three option types:
```
export declare enum OptionType {
    unspecified_option_type = "unspecified_option_type",
    drop_down = "drop_down",
    color = "color"
}
```

I'm rendering color options in one way and all other as a select dropdown... so, all options are rendered. unknown are redered as select.

regarding filtering valid options... I was thinking about that to have some util to filter out non-valid options (name or choices are undefined etc.), but this will make whole codebase more complex. also, I've never seen such case if something is undefined in options on practice. and backoffice UI does not allow to have such options, so I'm not sure why typings are done in this way for e-commerce sdk